qiskit_nature/results/utils/protein_shape_file_gen.py

```diff
@@ -0,0 +1,148 @@
+
+    def __init__(
+        self,
+        main_chain_turns: List[int],
+        side_chain_turns: List[Union[None, int]],
+        peptide: Peptide,
+    ) -> None:
+        """
+        Args:
+            main_chain_turns: a list of integers encoding the turns of the main chain.
+            side_chain_turns: a list of integers and None encoding the turns of the main chain
+                              or None.
+            peptide: the peptide we are getting the positions for.
+
+        """
+        self._main_chain_turns = main_chain_turns
+        self._side_chain_turns = side_chain_turns
+
+        self._main_chain_aminoacid_list = np.array(
+            list(peptide.get_main_chain.main_chain_residue_sequence)
+        )
+        self._side_chain_aminoacid_list = np.array(
+            [
+                aminoacid.residue_sequence[0] if (aminoacid is not None) else None
+                for aminoacid in peptide.get_side_chains()
+            ]
+        )
+        self._main_positions = self.generate_main_positions()
+        self._side_positions = self.generate_side_positions()
+
+    def generate_side_positions(self) -> List[Optional[np.ndarray]]:
+        """Generates the positions of the side chain."""
+        side_positions: List[Optional[np.ndarray]] = []
+        counter = 1
+        for mainpos, sideturn in zip(self.main_positions, self._side_chain_turns):
+            if sideturn is None:
+                side_positions.append(None)
+            else:
+                side_positions.append(mainpos + (-1) ** counter * self.coordinates[sideturn])
+
+            counter += 1
+        return side_positions
+
+    @property
+    def side_positions(self) -> List[Optional[np.ndarray]]:
+        """
+        Returns the xyz position for each side chain element.
+
+        Returns:
+            A list with the position of the side chain of each bead in the main chain in order.
+            None in the i-th position of the list corresponds to no side chain at that
+            position of the main chain.
+
+        """
+
+        return self._side_positions
+
+    def generate_main_positions(self) -> np.ndarray:
+        """Generates the positions of the main chain."""
+        length_turns = len(self._main_chain_turns)
+        relative_positions = np.zeros((length_turns + 1, 3), dtype=float)
+
+        for i in range(length_turns):
+            relative_positions[i + 1] = (-1) ** i * self.coordinates[self._main_chain_turns[i]]
+
+        return relative_positions.cumsum(axis=0)
+
+    @property
+    def main_positions(self) -> np.ndarray:
+        """
+        Returns an array with the cartesian coordinates of each aminoacid in the main chain.
+        The first time called it generates the coordinates.
+
+        Returns:
+            An array with the cartesian coordinates of each aminoacid in the main chain.
+
+        """
+
+        return self._main_positions
+
+    def save_xyz_file(self, name: str, path: str = "") -> None:
```

Comment:
I don't see a test which validates the produced XYZ files.

Also make sure that these are indeed valid files, e.g. I currently do not see an empty line being produced below the number of particles on the first line. (The second line of the file is a comment line and will be ignored by any parser)

Reply:
I had made a comment about the xyz format earlier too. The first line is number of atoms https://github.com/Qiskit/qiskit-nature/pull/655#issuecomment-1125039489